### PR TITLE
AnneJu step 11: Add model validations

### PR DESCRIPTION
**Pull request by anneju**
#### Description
- Add task model db validation
- Add gem `shoulda-matchers` to help with model validation

#### Stories
- [x] Add gem `shoulda-matchers`
- [x] Add db migration to regulate from db level
- [x] Write Specs

#### Related RSpecs
```
rspec spec/models/tasks_spec.rb
```

### CODE REVIEW

File: myapp/app/models/task.rb [hunk 1]
```diff
@@ -5,11 +5,11 @@
...
 #
 #  id          :bigint           not null, primary key
-#  title       :string(255)
-#  description :string(255)
+#  description :string(255)      not null
+#  due_date    :datetime
 #  priority    :integer
 #  status      :integer
-#  due_date    :datetime
+#  title       :string(255)      not null
```
Comment (anneju) [inline myapp/app/models/task.rb:12]: auto created by gem `annotate` (sometimes it move around the order of columns that I cannot control 😅)

File: myapp/db/migrate/20221221025136_change_tasks_title_and_description_null_false.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class ChangeTasksTitleAndDescriptionNullFalse < ActiveRecord::Migration[6.0]
+  def up
+    change_column :tasks, :title, :string, null: false
+    change_column :tasks, :description, :string, null: false
```
Comment (anneju) [inline myapp/db/migrate/20221221025136_change_tasks_title_and_description_null_false.rb:4]: I haven't set restrictions to other columns as I think those are to be implemented in upcoming steps (I only display `title` and `description` on my view now)

File: myapp/spec/models/tasks_spec.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+describe Task, type: :model do
+  describe 'validations' do
+    subject { build(:task) }
+
+    it { should validate_presence_of(:title) }
+    it { should validate_presence_of(:description) }
```
Comment (anneju) [inline myapp/spec/models/tasks_spec.rb:8]: only added model specs here as I have wrote the system specs for error messages in previous PR

https://github.com/Fablic/training/blob/d1415fdd35194ec03bc9347368dbc1c06c649b0c/myapp/spec/system/tasks_spec.rb#L91-L104

File: myapp/Gemfile.lock [hunk 4]
```diff
@@ -224,6 +224,8 @@ GEM
       rexml (~> 3.2, >= 3.2.5)
       rubyzip (>= 1.2.2)
     semantic_range (3.0.0)
+    shoulda-matchers (5.3.0)
```
Comment (stakahashi-d) [inline myapp/Gemfile.lock:227]: 👍

**Review by stakahashi-d — APPROVED** [review]
LGTM

**Pull request merged**